### PR TITLE
feat: [SRM-12851]: Updated reset single slo

### PR DESCRIPTION
**Pull request by deepesh-ui**
#### Screenshots

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx [hunk 1]
```diff
@@ -37,6 +38,40 @@ export const getDistribution = (
...
+        return clonedArr[num]?.weightagePercentage + total
+      }, 0)
+  const newWeight = remaining / (length - (freezedIndexs.length || 1))
+
+  for (let idx = 0; idx < originalList.length; idx++) {
+    if (currentIndex === idx) {
+      clonedArr[idx].weightagePercentage = Number(weight)
+      clonedArr[idx].isManuallyUpdated = true
+    } else if (currentIndex !== idx && freezedIndexs.includes(idx)) {
+      // don't update
```
Comment (praneshg239) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx:65]: do we need this comment?

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx [hunk 2]
```diff
@@ -104,13 +139,22 @@ export const onWeightChange = ({
...
-  if (weight < 100) {
-    const neweDist = getDistribution(weight, index, serviceLevelObjectivesDetails)
-    setServiceLevelObjectivesDetails(neweDist)
-  } else {
-    const cloneList = [...serviceLevelObjectivesDetails]
-    cloneList[index].weightagePercentage = weight
-    setServiceLevelObjectivesDetails(cloneList)
+  if (weight < 100 && /^\d+(?:\.\d{1,2})?$/.test(weight.toString())) {
+    const _serviceLevelObjectivesDetails = cloneDeep(serviceLevelObjectivesDetails)
+    // const neweDist = getDistribution(weight, index, _serviceLevelObjectivesDetails)
```
Comment (praneshg239) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx:144]: please remove the commented code

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx [hunk 3]
```diff
@@ -104,13 +139,22 @@ export const onWeightChange = ({
...
+    const neweDistInta = getDistributionUpdated(
```
Comment (praneshg239) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx:145]: NIT: If we have more parameters, we can pass them as objects.

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx [hunk 4]
```diff
@@ -129,6 +132,23 @@ export const AddSLOs = (): JSX.Element => {
           name="weightagePercentage"
           value={row.original.weightagePercentage.toString()}
         />
+        {serviceLevelObjectivesDetails[row.index]?.isManuallyUpdated && (
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx:135]: null check for row.index

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx [hunk 5]
```diff
@@ -129,6 +132,23 @@ export const AddSLOs = (): JSX.Element => {
...
+          <Button
+            icon="plus"
+            minimal
+            withoutCurrentColor
+            onClick={() => {
+              const resetWeight = dashboardWidgetsResponse?.data?.content ? 10 : 0
+              onWeightChange({
+                index: row.index,
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx:143]: same.

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx [hunk 6]
```diff
@@ -37,6 +38,40 @@ export const getDistribution = (
...
 }
 
+export const getDistributionUpdated = (
+  weight: number,
+  currentIndex: number,
+  originalList: SLOObjective[],
+  freezedIndexs: number[]
+): SLOObjective[] => {
+  const clonedArr = [...originalList]
+  let _freezedIndexs = cloneDeep(freezedIndexs)
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx:48]: use camel casing everywhere

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx [hunk 7]
```diff
@@ -37,6 +38,40 @@ export const getDistribution = (
...
+    100 -
+    weight -
+    freezedIndexs
+      .filter(item => item !== currentIndex)
+      .reduce((total, num) => {
+        return clonedArr[num]?.weightagePercentage + total
+      }, 0)
+  const newWeight = remaining / (length - (freezedIndexs.length || 1))
+
+  for (let idx = 0; idx < originalList.length; idx++) {
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx:60]: these can we wrapped to smaller functions which will tell us what this logic is doing

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx [hunk 8]
```diff
@@ -104,13 +139,22 @@ export const onWeightChange = ({
...
-    cloneList[index].weightagePercentage = weight
-    setServiceLevelObjectivesDetails(cloneList)
+  if (weight < 100 && /^\d+(?:\.\d{1,2})?$/.test(weight.toString())) {
+    const _serviceLevelObjectivesDetails = cloneDeep(serviceLevelObjectivesDetails)
+    // const neweDist = getDistribution(weight, index, _serviceLevelObjectivesDetails)
+    const neweDistInta = getDistributionUpdated(
+      weight,
+      index,
+      serviceLevelObjectivesDetails,
+      _serviceLevelObjectivesDetails
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx:149]: camel casing

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.tsx [hunk 9]
```diff
@@ -211,7 +212,7 @@ export const SLOList = ({ filter, onAddSLO, hideDrawer, serviceLevelObjectivesDe
           <Button
             width={150}
             data-testid={'addSloButton'}
-            disabled={selectedSlos.length < 2}
+            disabled={selectedSlos?.length < MinNumberOfSLO || selectedSlos?.length > MaxNumberOfSLO}
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.tsx:215]: move this logic to a variable `isDisabled  = selectedSlos?.length < MinNumberOfSLO || selectedSlos?.length > MaxNumberOfSLO`

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx (hunk 4 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx:135]: done

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx (hunk 5 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx:143]: row will have index as it is handed by TableV2

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx (hunk 6 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx:48]: done

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx (hunk 1 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx:65]: removed

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx (hunk 7 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx:60]: done

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx (hunk 8 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx:149]: done

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx (hunk 3 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.utils.tsx:145]: done

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.tsx (hunk 9 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.tsx:215]: done

**Review by harshilbits — APPROVED** [review]

**Pull request merged**